app/models/miq_worker.rb

```diff
@@ -219,6 +219,10 @@ def worker_settings(options = {})
     self.class.fetch_worker_settings_from_server(miq_server, options)
   end
 
+  def heartbeat_file
+    ENV["WORKER_HEARTBEAT_FILE"] || Rails.root.join("tmp", guid + ".hb")
```

Comment:
Wat!?  No string interpolation... but using string concatenation!  For shame... :trollface: